trainee_template/ablation.amlg

```diff
@@ -0,0 +1,363 @@
+            (or 
+                (not (and hasInfluenceWeightEntropies))
+                (< (call GetNumTrainingCases) autoAblateMinModelSize )
+            )
+            (conclude 1)
+        )
+
+        (assign (assoc
+            max_influence_weight_entropy_to_keep
+                (compute_on_contained_entities (list
+                    (query_exists internalLabelInfluenceWeightEntropy)
+                    (query_quantile internalLabelInfluenceWeightEntropy autoAblateInfluenceWeightEntropyThreshold ".case_weight")
+                ))
+        ))
+
+        (+
+            (or
+                (> max_influence_weight_entropy_to_keep
+                    (call ComputeInfluenceWeightEntropy (assoc
+                        context_values context_values
+                        context_features context_features
+                    ))
+                    
+                )
+                ; This is probably unnecessary but in case we cannot compute the requested influence weight
+                ; quantile return 1 as well.
+                (= max_influence_weight_entropy_to_keep .nan)
+            )
+        )
+    )
+
+    #CaseOutsideThresholds
+	(let
+		(assoc
+			action_features
+                (filter (lambda
+                    (or 
+                        (contains_value autoAblateExactPredictionFeatures (current_value))
+                        (contains_index autoAblateTolerancePredictionThresholdMap (current_value))
+                        (contains_index autoAblateRelativePredictionThresholdMap (current_value))
+                        (contains_value autoAblateResidualPredictionFeatures (current_value))
+                    )
+                ) features )
+			feature_value_map (zip features feature_values)
+		)
+
+        (if
+            (= (null)
+                autoAblateExactPredictionFeatures
+                autoAblateTolerancePredictionThresholdMap
+                autoAblateRelativePredictionThresholdMap
+                autoAblateResidualPredictionFeatures
+                autoAblateConvictionLowerThreshold
+                autoAblateConvictionUpperThreshold
+            )
+            (conclude 1)
+        )
+
+		(declare (assoc
+			action_values (unzip feature_value_map action_features)
+			residual_map (assoc)
+		))
+
+		(or ;evaluate in order and return the first true block
+
+			;if the model is too small, skip ablatement, return true to force training
+			(< (call GetNumTrainingCases) autoAblateMinModelSize)
+
+			;if we have action_features, need to react first
+			(if (!= (list) action_features)
+				;store the reaction to reuse for both
+				(let
+					(assoc
+						reaction_values
+							(map
+								(lambda (let
+									(assoc
+										feature (current_index 1)
+										;flag set if this feature uses residual as its threshold type
+										ablate_by_feature_residuals
+                                            (contains_index (zip autoAblateResidualPredictionFeatures) (current_index 1))
```

Comment:
`(contains_index (zip x) y)` should be the same as just `(contains_value x y)` as far as I know